caas/kubernetes/provider/pod.go

```diff
@@ -0,0 +1,199 @@
+// Copyright 2020 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package provider
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/juju/errors"
+
+	"github.com/juju/juju/core/status"
+
+	core "k8s.io/api/core/v1"
+)
+
+type EventGetter func() ([]core.Event, error)
+
+const (
+	ReasonCompleted                = "Completed"
+	ReasonContainersNotInitialized = "ContainersNotInitialized"
+	ReasonContainersNotReady       = "ContainersNotReady"
+	ReasonCrashLoopBackoff         = "CrashLoopBackOff"
+	ReasonError                    = "Error"
+	ReasonImagePull                = "ErrImagePull"
+)
+
+var (
+	podContainersReadyReasonsMap = map[string]status.Status{
+		ReasonContainersNotReady: status.Maintenance,
+	}
+
+	podInitializedReasonsMap = map[string]status.Status{
+		ReasonContainersNotInitialized: status.Maintenance,
+	}
+
+	podReadyReasonMap = map[string]status.Status{
+		ReasonContainersNotReady:       status.Maintenance,
+		ReasonContainersNotInitialized: status.Maintenance,
+	}
+
+	podScheduledReasonsMap = map[string]status.Status{
+		core.PodReasonUnschedulable: status.Blocked,
+	}
+)
+
+// GetPodCondition extracts the provided condition from the given status and returns that.
+// Returns nil and -1 if the condition is not present, and the index of the located condition.
+func GetPodCondition(status *core.PodStatus, conditionType core.PodConditionType) (int, *core.PodCondition) {
+	if status == nil {
+		return -1, nil
+	}
+	return GetPodConditionFromList(status.Conditions, conditionType)
+}
+
+// GetPodConditionFromList extracts the provided condition from the given list of condition and
+// returns the index of the condition and the condition. Returns -1 and nil if the condition is not present.
+func GetPodConditionFromList(conditions []core.PodCondition, conditionType core.PodConditionType) (int, *core.PodCondition) {
+	if conditions == nil {
```

Comment:
I think just combine these into a single method. They don't really pay their way having them split up.

Reply:
I have copied and pasted these methods from the Kubernetes code base.

Reply:
Done & Done